core/trino-main/src/test/java/io/trino/operator/scalar/FunctionAssertions.java

```diff
@@ -377,7 +377,7 @@ public void assertCachedInstanceHasBoundedRetainedSize(String projection)
                     newSimpleAggregatedMemoryContext().newLocalMemoryContext(PageProcessor.class.getSimpleName()),
                     SOURCE_PAGE);
             // consume the iterator
-            Optional<Page> ignored = Iterators.getOnlyElement(output);
+            Iterators.getOnlyElement(output);
```

Comment:
Can this be optimized away by the JIT? I don't know if it's equivalent.

Reply:
if JIT can prove removal is end-state invisible. However `getOnlyElement` does some checks, so JIT would first need to prove the checks are always satisifed.

Anyway, variable assignment doesn't change anything from this perspective.

Reply:
This didn't fail the `ReturnValueIgnored` Error Prone check? Maybe it got smarter in the mean time...

Reply:
`ReturnValueIgnored` only looks at a hardcoded list of methods (most of them from the JDK instead of libraries).
But yeah, I wish it was smarter.

Reply:
It used to fail on this particular line, that's why I added the `ignored` local variable.

(It also checks methods annotated with `@CheckReturnValue` or similar, but `getOnlyElement` does not have it.)